src/thunderbots/software/ai/primitive/stop_primitive.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "ai/primitive/primitive.h"
+
+class StopPrimitive : public Primitive
+{
+    public:
+     static const std::string PRIMITIVE_NAME;
+
+    /**
+     * Creates a new Stop Primitive
+     *
+     * @param robot_id The id of the Robot to run this Primitive
+     * @param stop to stop the robot or not
+     */
+     explicit StopPrimitive(unsigned int robot_id, bool coast);
+
+    /**
+     * Creates a new Stop Primitive from a Primitive message
+     *
+     * @param primitive_msg The message from which to create the Stop Primitive
+     */
+     explicit StopPrimitive(const thunderbots_msgs::Primitive &primitive_msg);
+
+     std::string getPrimitiveName() const override;
+
+     unsigned int getRobotId() const override;
+
+     bool coastRobot() const;
+
+     std::vector<double> getParameters() const override;
+
+     std::vector<bool> getExtraBits() const override;
```

Comment:
`getParameters` and `getExtraBits` should have descriptive javadoc comments, see other primitives for examples

Reply:
I don't think it needs one. This is overriding a function, and the commend documentation is there. We typically haven't been re-commenting overridden functions.

Reply:
I think it's important that these functions have explicit documentation for the form of the vector that they're returning (or if those vectors are empty) because it's otherwise not indicated what the semantic meaning of each element in the vector means. See other primitives for examples.

Reply:
Ok fair enough.